webui/react/src/pages/TrialDetails/TrialDetailsLogs.tsx

```diff
@@ -182,23 +205,171 @@ const TrialDetailsLogs: React.FC<Props> = ({ experiment, trial }: Props) => {
   const logFilters = (
     <LogViewerSelect
       options={filterOptions}
-      showSearch={true}
+      searchOn={searchOn}
+      showSearch={false}
       values={filterValues}
       onChange={handleFilterChange}
+      onClickSearch={() => setSearchOn((prev) => !prev)}
       onReset={handleFilterReset}
     />
   );
 
+  const throttledChangeSearch = useMemo(
+    () =>
+      throttle(
```

Comment:
nit: should this be debounce instead? that way we wait until the user is done typing before updating the settings while also allowing us to reduce the amount of time the user has to wait for the settings to commit.